### PR TITLE
(feat) Migrating Vitals and Biometrics form to RHF and Zod

### PR DESCRIPTION
**Pull request by lumuchris256**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Migrated the form to use reacthookform and zod 

## Screenshots
[Screencast from 2023-08-14 16-15-44.webm](https://github.com/openmrs/openmrs-esm-patient-chart/assets/59338693/764faf0c-7017-4a61-8aee-2f13ee5a232c)

## Related Issue
https://issues.openmrs.org/browse/O3-2235

## Other
cc @hadijahkyampeire @denniskigen

### CODE REVIEW

**Review by hadijahkyampeire — COMMENTED** [review]
Thanks @lumuchris256 , just wondering why you didn't validate single inputs, what happens if wrong values are input, does it throw any error or warning on the specific input.

**Comment by lumuchris256** [comment]
> > 
> 
> Validation is handled for every input 
> 
> vitalsvalidation

**Comment by lumuchris256** [comment]
> Thanks @lumuchris256 , just wondering why you didn't validate single inputs, what happens if wrong values are input, does it throw any error or warning on the specific input.

Validations to single inputs are present ,needed some improvements 
chnages ready for re-review

[Screencast from 2023-08-27 13-52-05.webm](https://github.com/openmrs/openmrs-esm-patient-chart/assets/59338693/f8491d98-b5bc-43eb-8af0-42f6885bbf73)

File: packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-form.test.tsx [hunk 1]
```diff
@@ -146,36 +146,6 @@ describe('VitalsBiometricsForm: ', () => {
 
     await waitFor(() => user.click(saveButton));
 
-    expect(mockSavePatientVitals).toHaveBeenCalledTimes(1);
```
Comment (hadijahkyampeire) [inline packages/esm-patient-vitals-app/src/vitals/vitals-biometrics-form/vitals-biometrics-form.test.tsx:149]: Why did you remove these tests, we need to test that the form submits successfully.

**Review by hadijahkyampeire — CHANGES_REQUESTED** [review]
@lumuchris256 I am not sure why you removed most of the tests but we need them, Let me know why you you had to remove them.

**Comment by jayasanka-sack** [comment]
@lumuchris256 @hadijahkyampeire I have to make the same comment as here: https://github.com/openmrs/openmrs-esm-patient-chart/pull/1278#issuecomment-1717498642

**Pull request merged**